src/assets/CollectiblesController.ts

```diff
@@ -1148,6 +1150,97 @@ export class CollectiblesController extends BaseController<
 
     this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
   }
+
+  /**
+   * Returns the collectible by the address and token id
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @returns a collectible
+   */
+  findCollectibleByAddressAndTokenId(
+    address: string,
+    tokenId: string,
+  ): Collectible | undefined {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address.toLowerCase() === address.toLowerCase() &&
+        collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+
+    return collectibles[index];
+  }
+
+  /**
+   * Update collectible data
+   *
+   * @param address - Hex address of the collectible contract.
+   * @param tokenId - Number that represents the id of the token.
+   * @param collectible - Hex id of the send transaction of the collectible.
+   */
+  updateCollectibleByAddressAndTokenId(
+    address: string,
+    tokenId: string,
+    collectible: Collectible,
+  ) {
+    const { allCollectibles } = this.state;
+    const { chainId, selectedAddress } = this.config;
+    const collectibles = allCollectibles[selectedAddress]?.[chainId] || [];
+    const index: number = collectibles.findIndex(
+      (collectible) =>
+        collectible.address === address && collectible.tokenId === tokenId,
+    );
+
+    if (index === -1) {
+      return;
+    }
+
+    const updatedCollectible: Collectible = {
+      ...collectible,
+    };
+
+    // Update Collectibles array
+    collectibles[index] = updatedCollectible;
```

Comment:
This will modify `collectibles` in state directly. You probably don't want that to happen just yet — you probably want to let `updateNestedCollectibleState` handle updating the state — so what if you do this instead:

```
const newCollectibles = [
  ...collectibles.slice(0, index),
  updatedCollectible,
  ...collectibles.slice(index + 1)
];
this.updateNestedCollectibleState(collectibles, ALL_COLLECTIBLES_STATE_KEY);
```